crawler.py

```diff
@@ -1,70 +1,93 @@
 import mechanize
-from collections import namedtuple
+import sys
+import pprint
+import collections
 from urllib.parse import urljoin, urlparse
 
-def validated(url):
-    '''Checks if given url is proper.'''
-    browser = mechanize.Browser()
-    browser.set_handle_robots(False)
-    try:
-        browser.open(url)
+
+def is_validated(url):
+    """Checks if given url is proper."""
+    parsed_url = urlparse(url)
+    if parsed_url.scheme and parsed_url.netloc:
```

Comment:
You can leverage pythons flexibility here and simply have

```python
return parsed_url.scheme and parsed_url.netloc
```
`and` operator tells interpreter that it needs to check for truth value of first and then second argument.
https://docs.python.org/3.7/library/stdtypes.html#boolean-operations-and-or-not

Reply:
This way just made my day, amazing!